hack/verify-flags-underscore.py

```diff
@@ -0,0 +1,213 @@
+        flags.add(line)
+    f.close()
+
+    regexs = [ re.compile('Var[P]?\([^,]*, "([^"]*)"'),
+               re.compile('.String[P]?\("([^"]*)",[^,]+,[^)]+\)'),
+               re.compile('.Int[P]?\("([^"]*)",[^,]+,[^)]+\)'),
+               re.compile('.Bool[P]?\("([^"]*)",[^,]+,[^)]+\)'),
+               re.compile('.Duration[P]?\("([^"]*)",[^,]+,[^)]+\)'),
+               re.compile('.StringSlice[P]?\("([^"]*)",[^,]+,[^)]+\)') ]
+
+    new_flags = []
+    # walk all the files looking for any flags being declared
+    for pathname in files:
+        if not pathname.endswith(".go"):
+            continue
+        f = open(pathname, 'r')
+        data = f.read()
+        f.close()
+        matches = []
+        for regex in regexs:
+            matches = matches + regex.findall(data)
+        for flag in matches:
+            # if the flag doesn't have a - or _ it is not interesting
+            if not dashRE.search(flag):
+                continue
+            if flag not in flags:
+                new_flags.append(flag)
+    if len(new_flags) != 0:
+        print("Found flags in golang files not in the list of known flags. Please add these to hack/verify-flags/known-flags.txt")
+        print("%s" % "\n".join(new_flags))
+        sys.exit(1)
+    return list(flags)
+
+def flags_to_re(flags):
+    """turn the list of all flags we found into a regex find both - and _ version"""
+    flagREs = []
+    for flag in flags:
+        # turn all flag names into regexs which will find both types
+        newre = dashRE.sub('[-_]', flag)
+        flagREs.append(newre)
+    # turn that list of regex strings into a single large RE
+    flagRE = "|".join(flagREs)
+    flagRE = re.compile(flagRE)
+    return flagRE
+
+def load_exceptions(rootdir):
+    exceptions = {}
+    if args.skip_exceptions:
+        return exceptions
+    exception_filename = os.path.join(rootdir, "hack/verify-flags/exceptions.txt")
+    exception_file = open(exception_filename, 'r')
+    for exception in exception_file.read().splitlines():
+        out = exception.split(":", 1)
+        if len(out) != 2:
+            printf("Invalid line in exceptions file: %s" % exception)
+            continue
+        filename = out[0]
+        line = out[1]
+        exceptions[(filename, line)] = True
+    return exceptions
+
+def main():
+    rootdir = os.path.dirname(__file__) + "/../"
+    rootdir = os.path.abspath(rootdir)
+
+    exceptions = load_exceptions(rootdir)
+
+    if len(args.filenames) > 0:
+        files = args.filenames
+    else:
+        files = get_all_files(rootdir)
+    files = normalize_files(rootdir, files)
+
+    flags = get_flags(rootdir, files)
+    flagRE = flags_to_re(flags)
+
+    bad_lines = []
+    # walk all the file looking for any flag that was declared and now has an _
+    for pathname in files:
+        relname = os.path.relpath(pathname, rootdir)
+        f = open(pathname, 'r')
+        for line in f.read().splitlines():
```

Comment:
Just to be certain, this is fine to operate on individual lines because we're matching explicit flag names and they shouldn't split across lines anyways?